uml/design.uml

```diff
@@ -86,14 +113,52 @@ class Minion {
     damage_dealt
     type
     abilities
+    if_played_this_turn
 
     attack_minion(Minion)
     attack_hero(Hero)
 }
 
 class Spell {
+    Action
+
     cast()
 }
 
+class Minion_abilities {
```

Comment:
If I got this correctly – Minion has MinionAbilities object which fields can be:
 - boolean values e.g. charge = True
 - some triggered event(consequence) - e.g. battlecry -> "deal 2 damage to all minions"